src/main/java/uk/gov/hmcts/darts/arm/service/impl/AbstractArmBatchProcessResponseFiles.java

```diff
@@ -0,0 +1,836 @@
+                                                       ArmResponseUploadFileRecord armResponseUploadFileRecord,
+                                                       CreateRecordFilenameProcessor createRecordFilenameProcessor) {
+        log.warn("Unable to find external object directory with ID {} for ARM batch responses with CR file {}, UF file {}",
+                 externalObjectDirectoryId,
+                 createRecordFilenameProcessor.getCreateRecordFilenameAndPath(),
+                 uploadFileFilenameProcessor.getUploadFileFilenameAndPath());
+        List<String> validResponseFiles = List.of(createRecordFilenameProcessor.getCreateRecordFilenameAndPath(),
+                                                  uploadFileFilenameProcessor.getUploadFileFilenameAndPath());
+        deleteResponseBlobs(validResponseFiles);
+        log.warn(
+            "Unable to process upload file {} with EOD record {}, IU file {}", uploadFileFilenameProcessor.getUploadFileFilenameAndPath(),
+            armResponseUploadFileRecord.getA360RecordId(), armResponseUploadFileRecord.getA360FileId());
+    }
+
+
+    private void processUploadFileDataSuccess(BatchInputUploadFileFilenameProcessor batchUploadFileFilenameProcessor,
+                                              ArmResponseBatchData armResponseBatchData,
+                                              ExternalObjectDirectoryEntity externalObjectDirectory,
+                                              ArmResponseUploadFileRecord armResponseUploadFileRecord) {
+        if (externalObjectDirectory.getChecksum() != null) {
+            verifyChecksumAndUpdateStatus(batchUploadFileFilenameProcessor,
+                                          armResponseBatchData,
+                                          armResponseUploadFileRecord,
+                                          externalObjectDirectory,
+                                          externalObjectDirectory.getChecksum());
+        } else {
+            log.warn("Unable to verify checksum for external object {}", externalObjectDirectory.getId());
+            updateExternalObjectDirectoryStatus(externalObjectDirectory, EodHelper.armResponseChecksumVerificationFailedStatus());
+        }
+    }
+
+    private void verifyChecksumAndUpdateStatus(BatchInputUploadFileFilenameProcessor batchUploadFileFilenameProcessor,
+                                               ArmResponseBatchData armResponseBatchData,
+                                               ArmResponseUploadFileRecord armResponseUploadFileRecord,
+                                               ExternalObjectDirectoryEntity externalObjectDirectory,
+                                               String objectChecksum) {
+        if (objectChecksum.equalsIgnoreCase(armResponseUploadFileRecord.getMd5())) {
+            onUfChecksumValidationSuccess(batchUploadFileFilenameProcessor,
```

Comment:
author note: refactoring, existing logic moved to `onUfChecksumValidationSuccess`